Unit tests fail on i386

Unit tests fail with healpy 1.12.8 on Debian i386. See [full build log](https://buildd.debian.org/status/fetch.php?pkg=healpy&arch=i386&ver=1.12.8-4&stamp=1547914020&file=log) and excerpt of the most relevant part below.

```
=================================== FAILURES ===================================
______________________ [doctest] healpy.pixelfunc.ang2pix ______________________
440     --------
441     pix2ang, pix2vec, vec2pix
442 
443     Examples
444     --------
445     >>> import healpy as hp
446     >>> hp.ang2pix(16, np.pi/2, 0)
447     1440
448 
449     >>> print(hp.ang2pix(16, [np.pi/2, np.pi/4, np.pi/2, 0, np.pi], [0., np.pi/4, np.pi/2, 0, 0]))
Expected:
    [1440  427 1520    0 3068]
Got:
    [1440  427 1456    0 3068]

/<<PKGBUILDDIR>>/.pybuild/cpython3_3.7_healpy/build/healpy/pixelfunc.py:449: DocTestFailure
________________ [doctest] healpy.pixelfunc.get_interp_weights _________________
1334     array([ 0.25,  0.25,  0.25,  0.25])
1335 
1336     >>> pix, weights = hp.get_interp_weights(1, 0, 90, lonlat=True)
1337     >>> print(pix)
1338     [1 2 3 0]
1339     >>> weights
1340     array([ 0.25,  0.25,  0.25,  0.25])
1341 
1342     >>> pix, weights = hp.get_interp_weights(1, [0, np.pi/2], 0)
1343     >>> print(pix)
Differences (unified diff with -expected +actual):
    @@ -1,4 +1,4 @@
    -[[ 1  4]
    - [ 2  5]
    - [ 3 11]
    - [ 0  8]]
    +[[1 3]
    + [2 0]
    + [3 4]
    + [0 5]]
```

--- a/healpy/pixelfunc.py
+++ b/healpy/pixelfunc.py
@@ -442,14 +442,19 @@ def ang2pix(nside, theta, phi, nest=False, lonlat=False):
 
     Examples
     --------
+    Note that some of the test inputs below that are on pixel boundaries
+    such as theta=pi/2, phi=pi/2, have a tiny value of 1e-15 added to them
+    to make them reproducible on i386 machines using x87 floating point
+    instruction set (see https://github.com/healpy/healpy/issues/528).
+
     >>> import healpy as hp
     >>> hp.ang2pix(16, np.pi/2, 0)
     1440
 
-    >>> print(hp.ang2pix(16, [np.pi/2, np.pi/4, np.pi/2, 0, np.pi], [0., np.pi/4, np.pi/2, 0, 0]))
+    >>> print(hp.ang2pix(16, [np.pi/2, np.pi/4, np.pi/2, 0, np.pi], [0., np.pi/4, np.pi/2 + 1e-15, 0, 0]))
     [1440  427 1520    0 3068]
 
-    >>> print(hp.ang2pix(16, np.pi/2, [0, np.pi/2]))
+    >>> print(hp.ang2pix(16, np.pi/2, [0, np.pi/2 + 1e-15]))
     [1440 1520]
 
     >>> print(hp.ang2pix([1, 2, 4, 8, 16], np.pi/2, 0))
@@ -1320,6 +1325,11 @@ def get_interp_weights(nside, theta, phi=None, nest=False, lonlat=False):
 
     Examples
     --------
+    Note that some of the test inputs below that are on pixel boundaries
+    such as theta=pi/2, phi=pi/2, have a tiny value of 1e-15 added to them
+    to make them reproducible on i386 machines using x87 floating point
+    instruction set (see https://github.com/healpy/healpy/issues/528).
+
     >>> import healpy as hp
     >>> pix, weights = hp.get_interp_weights(1, 0)
     >>> print(pix)
@@ -1339,17 +1349,18 @@ def get_interp_weights(nside, theta, phi=None, nest=False, lonlat=False):
     >>> weights
     array([ 0.25,  0.25,  0.25,  0.25])
 
-    >>> pix, weights = hp.get_interp_weights(1, [0, np.pi/2], 0)
+    >>> pix, weights = hp.get_interp_weights(1, [0, np.pi/2 + 1e-15], 0)
     >>> print(pix)
     [[ 1  4]
      [ 2  5]
      [ 3 11]
      [ 0  8]]
-    >>> weights
-    array([[ 0.25,  1.  ],
-           [ 0.25,  0.  ],
-           [ 0.25,  0.  ],
-           [ 0.25,  0.  ]])
+    >>> np.testing.assert_allclose(
+    ...     weights,
+    ...     np.array([[ 0.25,  1.  ],
+    ...               [ 0.25,  0.  ],
+    ...               [ 0.25,  0.  ],
+    ...               [ 0.25,  0.  ]]), rtol=0, atol=1e-14)
     """
     check_nside(nside, nest=nest)
     if phi is None:

